### PR TITLE
Using invalidateChildInParent for API levels lower than 26.

### PR DESCRIPTION
**Pull request by 0xZOne**
Using `invalidateChildInParent` for API levels lower than 26.

/cc @blasten @dnfield 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java [hunk 1]
```diff
@@ -175,12 +177,20 @@ public boolean onInterceptTouchEvent(@NonNull MotionEvent event) {
     return true;
   }
 
+  /** Used on Android O+, {@link invalidateChildInParent} used for previous versions. */
+  @SuppressLint("NewApi")
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:181]: why is this required? Do you set a lower targetSdkVersion?

Comment (0xZOne) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:181]: I think we should also consider the API levels lower than 26. :)

Comment (dnfield) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:181]: Shouldn't this be a TargetApi attribute instead of suppress lint?

**Comment by dnfield** [comment]
Needs a test.

**Comment by 0xZOne** [comment]
> Needs a test.

Sorry, I don't know how to write tests for this situation. If there are guidelines, I'd be happy to add them. @dnfield  @blasten

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java (hunk 1 above)
Comment (dnfield) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewWrapper.java:181]: Yes, but the right way to express this to the linter is via `@TargetApi` and/or `@RequiresApi` - see https://stackoverflow.com/questions/40007365/requiresapi-vs-targetapi-android-annotations/45773141. SuppressLint works but it looks like we're ignoring a lint without explaining why. It's not a huge deal either way in this case, especially given the comment - but with those lints you don't need the comments since it's encoded in the attribute :)

**Comment by dnfield** [comment]
In `shell/platform/android/test/io/flutter/plugin/platform/PlatformViewWrapperTest.java`, add a test that verifies `invalidate()` gets called when you call `invalidateChildInParent` (which would fail without this patch).

**Comment by 0xZOne** [comment]
> In `shell/platform/android/test/io/flutter/plugin/platform/PlatformViewWrapperTest.java`, add a test that verifies `invalidate()` gets called when you call `invalidateChildInParent` (which would fail without this patch).

@dnfield Thank you for your patient reply.

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
@blasten can you provide a second review?

**Review by blasten — APPROVED** [review]
LGTM
